src/v/cluster/partition.cc

```diff
@@ -1838,6 +1844,13 @@ ss::future<result<ss::rwlock::holder>> partition::hold_writes_enabled() {
     co_return *std::move(maybe_units);
 }
 
+std::optional<int64_t> partition::cloud_topic_max_gc_eligible_epoch() const {
```

Comment:
I wish we didn't have these circular dependencies in the system - like I want cloud topics L0 to sit entirely ontop of cluster::partition, but I don't think that's realistic.

Reply:
yeh. i think unless someone can clearly "see" how it should all fit together ideally, i think we'd need dedicated time to unravel the knot.